Add cookiecutter structure to `/ds`

georgerichardson opened the pull request:
Set up cookiecutter with modifications outlined in #118 

closes #118

georgerichardson committed:
Add data science cookiecutter directory structure.

georgerichardson committed:
Remove GitHub setup

mindrones commented on ds/.github/pull_request_template.md:
```diff
@@ -0,0 +1,14 @@
+---
```
I think this file needs to be moved to the `.github` dir in the root of the repo

mindrones commented on ds/.pre-commit-config.yaml:
```diff
@@ -0,0 +1,61 @@
+repos:
```
Not sure if this one would work in this directory.

OTOH if it doesn't and you have to move it to the root, I don't think it would apply to `ui/`

georgerichardson replied:
Will it cause issues for you if it's in the root?

bishax replied:
If you add the following to the top of the file it should be fine

```yaml
files: ^ds/
exclude: data/
```

bishax replied:
You could remove the `exclude: data/` if you're willing to have a commit formatting the followiing 
![image](https://user-images.githubusercontent.com/15673462/122885429-fbbb8f80-d336-11eb-97a9-1cb1b85dafac.png)

mindrones commented on ds/LICENSE:
```diff
@@ -0,0 +1,10 @@
+
+The MIT License (MIT)
```
This file is redundant, as there is one in the root already

mindrones commented on ds/Makefile:
```diff
@@ -0,0 +1,195 @@
+SHELL := /bin/bash
+
```
Are we sure this script won't affect files outside of `ds/`?

georgerichardson replied:
I ran `make install` in a test branch and it had no effects outside of `ds/` so I think we're ok

mindrones commented on ds/bin/create_repo.sh:
```diff
@@ -0,0 +1,23 @@
+#!/bin/bash
+set -euo pipefail
```
do we need this file?

georgerichardson replied:
I don't think we do once the structure is setup. @bishax ?

bishax replied:
Your repo already exists so you don't need it.

mindrones commented on ds/docs/license.rst:
```diff
@@ -0,0 +1 @@
+.. include:: ../LICENSE
```
not sure about this file once we remove `ds/LICENSE`

mindrones commented on ds/inputs/README.md:
```diff
@@ -0,0 +1 @@
+Could this directory just be `inputs`?
```
Do you mean `/inputs` rather than `/ds/inputs`? (I'd prefer not)

georgerichardson replied:
Think this is a note from @bishax to himself :)

mindrones reviewed (changes requested):
Some changes seem required. I added some questions too.

georgerichardson committed:
Move PR template to root .github

georgerichardson committed:
Remove sub-directory license

georgerichardson committed:
Reference root license

georgerichardson committed:
Remove note

georgerichardson committed:
Remove reference to setup-github

bishax commented on ds/.gitignore:
```diff
@@ -0,0 +1,123 @@
+*.lock
+
```
```suggestion

.conda_exists
```

georgerichardson committed:
Add conda_exists to gitignore

georgerichardson committed:
Move to root and add file pattern for ds subdir

mindrones commented on .github/pull_request_template.md:
```diff
@@ -0,0 +1,14 @@
+---
```
Would it be possible to make sure this won't apply to commits not releative to `/ds`?

georgerichardson replied:
I think the new configuration should give the user the choice when making a PR

mindrones reviewed (commented):
Forgot, `ds/outputs/.gitkeep` isn't needed as there's now `ds/outputs/data`

georgerichardson committed:
Add PULL_REQUEST_TEMPLATE dir and ds specific template

- Added a PR template directory to allow for multiple template.
- Renamed the existing template to be data science specific

mindrones reviewed (approved):
Good to merge
